dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs

```diff
@@ -608,29 +602,46 @@ private static void TrackUniqueParameterType(ref bool hasParameterType, MethodIn
                         return null!;
                     }
 
-                    return Enum.Parse(targetType, input, ignoreCase: true);
-                };
-            }
+                    // Issue #1. Quick fix to prevent the exception thrown by ConvertTo method - System.ArgumentOutOfRangeException : Value cannot be null. (Parameter 'destinationType') (Parameter 'actual')
+                    // Context: UT - KernelFunctionTests2.ItSupportsArgumentsAsIsWithoutConvertingTheirTypeAsync, Line - await AssertParameterType<string?>(null);
+                    // For some reason, the target type that is "string?" in the method signature comes here as a string, and the above statement - 'if (wasNullable && input is null)' is not working.
+                    if (input is null)
+                    {
+                        return null;
+                    }
 
-            // Finally, look up and use a type converter. Again, special-case null if it was actually Nullable<T>.
-            if (TypeConverterFactory.GetTypeConverter(targetType) is TypeConverter converter && converter.CanConvertFrom(typeof(string)))
-            {
-                return (input, cultureInfo) =>
-                {
-                    if (wasNullable && input is null)
+                    // Issue? #2. Quick fix to prevent the exception thrown by ConvertTo method - System.ArgumentOutOfRangeException : 'EnumConverter' is unable to convert 'System.DayOfWeek' to 'System.DayOfWeek'. (Parameter 'f')
+                    // Context: UT - KernelFunctionTests2.ItSupportsConvertingFromManyTypesAsync, Line - FunctionResult result = await function.InvokeAsync(this._kernel, arguments);
+                    // I think this is helpful logic and it should remain. At least I have not identified any drawbacks of it except of the exception above.
+                    if (targetType == input?.GetType())
```

Comment:
The if block was added before previous one that checks for null and the `?` was never removed. Fixed now.

Reply:
I had to reintroduce the "?" because it is still possible for the input to be null. It is the only case when null is passed as an argument for a non-nullable parameter of a ValueType. In this case, it makes sense to propagate the null to a type converter and let it fails with a descriptive exception.